wallets/payment.js

```diff
@@ -0,0 +1,156 @@
+import { useCallback, useMemo } from 'react'
+import { decode as bolt11Decode } from 'bolt11'
+import { useWallets } from '@/wallets'
+import walletDefs from '@/wallets/client'
+import { formatSats } from '@/lib/format'
+import { useWalletLogger } from '@/components/wallet-logger'
+import { useInvoice } from '@/components/payment'
+import { FAST_POLL_INTERVAL } from '@/lib/constants'
+import { NoWalletAvailableError, SenderError, WalletAggregateError, WalletNotEnabledError } from '@/wallets/errors'
+
+export function useWalletPayment () {
+  const { wallets } = useWallets()
+  const invoiceHelper = useInvoice()
+
+  // XXX calling hooks in a loop is against the rules of hooks
+  //
+  // we do this here anyway since we need the logger for each wallet.
+  // we ensure hooks are always called in the same order by sorting the wallets by name.
+  //
+  // we don't use the return value of useWallets here because it is empty on first render
+  // so using it would change the order of the hooks between renders.
+  //
+  // see https://react.dev/reference/rules/rules-of-hooks
+  const loggers = walletDefs
+    .sort((def1, def2) => def1.name.localeCompare(def2.name))
```

Comment:
mhhh, I don't really need this `sort` since I can depend on `walletDefs` not changing it's order :thinking: